3-Register/index.html

```diff
@@ -0,0 +1,235 @@
+            const availableDomains = ['@gmail.com', '@hotmail.com', '@lightit.io']
+            const mailHasAcceptedDomain = availableDomains.some(domain => mail.endsWith(domain))
+
+            if (! mailHasAcceptedDomain) {
+                return populateErrors({ email: 'Email ends with invalid domain' })
+            }
+
+            return true
+        }
+
+        const usernameIsValid = function (username) {
+            const allowedChars = /^[a-z]+$/
+
+            if (username.length < 3) {
+                return populateErrors({ username: 'Usernames must have 3 or more characters' })
+            }
+
+            if (!allowedChars.test(username)) {
+                return populateErrors({ username: 'Usernames can only contain lowercase characters' })
+            }
+
+            return true
+        }
+
+        const passwordIsValid = function (password) {
+            const correctLength = password.length >= 8
+            const hasUppercase = /[A-Z]/.test(password)
+            const hasLowecase = /[a-z]/.test(password)
+            const hasSpecialChar = /[!@#$%^&*()_+{}[\]:;<>,.?~\\/-]/.test(password)
+
+            if (!correctLength) {
+                return populateErrors({ password: 'Password must be 8 characters or more'})
+            }
+
+            if (!hasUppercase) {
+                return populateErrors({ password: 'Password must have at least one uppercase character'})
+            }
+
+            if (!hasLowecase) {
+                return populateErrors({ password: 'Password must have at least one lowercase character'})
+            }
+
+            if (!hasSpecialChar) {
+                return populateErrors({ password: 'Password must have at least one special character'})
+            }
+
+            return true
+        }
+
+        const passwordAndConfirmMatch = function () {
+            const pswd = document.querySelector('#password-input')
+            const confirm = document.querySelector('#password-confirm-input')
+
+            if (pswd.value !== confirm.value) {
+                return populateErrors({ confirm: 'Password & Confirm Password fields do not match' })
+            }
+
+            return true
+        }
+
+        const populateErrors = function (errors) {
+            const emailError = document.querySelector('#email-error')
+            const usernameError = document.querySelector('#username-error')
+            const pswdError = document.querySelector('#password-error')
+            const confirmError = document.querySelector('#password-confirm-error')
+
+            if (errors.hasOwnProperty('email')) emailError.textContent = errors.email
+            if (errors.hasOwnProperty('username')) usernameError.textContent = errors.username
+            if (errors.hasOwnProperty('password')) pswdError.textContent = errors.password
+            if (errors.hasOwnProperty('confirm')) confirmError.textContent = errors.confirm
+
+            return false
+        }
```

Comment:
This function always returns `false`. I think it doesn't make sense for a function to return something if it's always the same thing.

It's like having a parameter that always has the same value, you could just delete the parameter and use that value.

We should remove the `return false`.